R/connect.R

```diff
@@ -0,0 +1,127 @@
+#' Class representing a Connect API client
+#'
+#' @name Client
+#'
+#' @section Details:
+#'
+#' This class allows a user to interact with a Connect server via the Connect
+#' API. Authentication is done by providing an API key.
+#'
+#' @importFrom utils capture.output
+#'
+#' @family R6 classes
+Client <- R6::R6Class( # nolint
+  "Client",
+  public = list(
+    server = NULL,
+    api_key = NULL,
+    minimum_server_version = "1.8.8.3",
+    initialize = function(server, api_key) {
+      self$server <- server
+      self$api_key <- api_key
+
+      private$validate()
+    },
+    print = function(...) {
+      cat("RStudio Connect API Client: \n")
+      cat("  Server: ", self$server, "\n", sep = "")
+      cat("  API Key: ", paste0(strrep("*", 11), substr(
+        self$api_key, nchar(self$api_key) - 3, nchar(self$api_key)
+      )), "\n", sep = "")
+      invisible(self)
+    },
+    GET = function(path, ..., writer = httr::write_memory(), parser = "text") {
+      url <- paste0(self$server, "/__api__", path)
+      res <- httr::GET(
+        url,
+        private$add_auth(),
+        writer,
+        ...
+      )
+      private$raise_error(res)
+      httr::content(res, as = parser)
+    },
+    content = function() {
+      results <- self$GET("/v1/content", query = list(include = "tags,owner"))
```

Comment:
I pulled this directly from connectwidgets. It may need to be audited so it only provides what we need. Or we can pivot to a one-off `httr` call to the server API